src/javascript/resolve-specifier-node.ts

```diff
@@ -67,26 +67,20 @@ export const resolve =
             },
       });
 
-      let relativeSpecifierUrl =
-          relative(dirname(documentPath), resolvedSpecifier) as FileRelativeUrl;
-
-      // If this is a component-style URL layout, rewrite the path to reach out
-      // of the current package, rather than directly into the component
-      // directory.
+      let effectiveDocumentPath = documentPath;
       if (componentInfo !== undefined) {
         const {packageName, rootDir, componentDir} = componentInfo;
-        if (pathIsInside(resolvedSpecifier, componentDir)) {
-          const componentDirRelativeToPackage =
-              relative(join(rootDir, packageName), rootDir);
-          const relativePathFromComponentDir =
-              relative(componentDir, resolvedSpecifier);
-          relativeSpecifierUrl =
-              join(
-                  componentDirRelativeToPackage,
-                  relativePathFromComponentDir) as FileRelativeUrl;
+        const isOwnPackageRequest = !pathIsInside(documentPath, componentDir);
+        if (isOwnPackageRequest) {
+          effectiveDocumentPath =
```

Comment:
let's add a comment here explaining the component URL mapping. Something like "component requests map root package URLs so that they appear to be (siblings of dependencies | paths within the components folder)."

Reply:
Done.